modules/auxiliary/gather/snare_registry.rb

```diff
@@ -0,0 +1,184 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit4 < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+
+  HttpFingerprint = { :pattern => [ /SNARE/ ] }
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'          => 'Snare Lite for Windows Registry Access',
+      'Description'   => %q{
+          This module uses the Registry Dump feature of the Snare Lite
+        for Windows service on 6161/TCP to retrieve the Windows registry.
+        The Dump Registry functionality is unavailable in Snare Enterprise.
+
+        Note: The Dump Registry functionality accepts only one connected
+        client at a time. Requesting a large key/hive will cause the service
+        to become unresponsive until the server completes the request.
+      },
+      'Platform'      => 'win',
+      'Author'        => [ 'Brendan Coles <bcoles[at]gmail.com>' ],
+      'License'       => MSF_LICENSE,
+      'References'    =>
+        [
+          [ 'URL', 'https://www.intersectalliance.com/wp-content/uploads/user_guides/Guide_to_Snare_for_Windows-4.2.pdf' ]
+        ],
+      'Actions'       =>
+        [
+          [ 'System Information', 'Description' => 'Retrieve information about the system' ],
+          [ 'Snare Information', 'Description' => 'Retrieve information about the Snare installation' ],
+          [ 'Dump Registry Key', 'Description' => 'Retrieve a specified registry key, including all sub-keys' ],
+          [ 'Dump Registry Hive', 'Description' => 'Retrieve a specified registry hive, including all sub-keys' ],
+          [ 'Dump Registry', 'Description' => 'Retrieve the entire Windows registry. (Note: this may take a while)' ]
+        ],
+      'DefaultAction' => 'System Information'
```

Comment:
These are good suggestions. My design design is hard to justify in retrospect. I'll update the PR shortly.